controller-logger-starter/src/main/java/com/example/controllerloggerstarter/logger/ControllerLogger.java

```diff
@@ -0,0 +1,89 @@
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.context.request.RequestContextHolder;
+import org.springframework.web.context.request.ServletRequestAttributes;
+
+import java.lang.annotation.Annotation;
+import java.lang.reflect.Method;
+import java.util.Enumeration;
+import java.util.Objects;
+
+import static com.example.controllerloggerstarter.constants.LoggerConstants.REQUEST_MESSAGE;
+import static com.example.controllerloggerstarter.constants.LoggerConstants.RESPONSE_MESSAGE;
+
+@RequiredArgsConstructor
+@Configuration
+@Aspect
+@Slf4j
+public class ControllerLogger {
+
+    private final ObjectMapper objectMapper;
+
+    @Around("@within(org.springframework.web.bind.annotation.RestController)")
+    public Object logRequest(ProceedingJoinPoint joinPoint) throws Throwable {
+        HttpServletRequest request = ((ServletRequestAttributes) Objects.requireNonNull(RequestContextHolder.getRequestAttributes())).getRequest();
+        String requestBody = getRequestBody(joinPoint);
+
+        log.info(REQUEST_MESSAGE,
+                request.getMethod(),
+                request.getRequestURI(),
+                getHeaders(request),
+                requestBody);
+
+        return getResponse(joinPoint, request);
+    }
+
+    private Object getResponse(ProceedingJoinPoint joinPoint, HttpServletRequest request) throws Throwable {
+        Object result = joinPoint.proceed();
+
+        log.info(RESPONSE_MESSAGE,
+                request.getMethod(),
+                request.getRequestURI(),
+                objectMapper.writeValueAsString(result));
+
+        return result;
+    }
+
+    private String getRequestBody(ProceedingJoinPoint joinPoint) {
+        MethodSignature methodSignature = (MethodSignature) joinPoint.getSignature();
+        Method method = methodSignature.getMethod();
+        Object[] args = joinPoint.getArgs();
+        int argLength = args.length;
+        for (int i = 0; i < argLength; i++) {
+            if (method.getParameterAnnotations()[i] != null) {
+                for (Annotation annotation : method.getParameterAnnotations()[i]) {
+                    if (annotation instanceof RequestBody) {
+                        try {
+                            return objectMapper.writeValueAsString(args[i]);
+                        } catch (Exception e) {
+                            log.error(e.getMessage());
+                        }
+                    }
+                }
+            }
+        }
+        return "";
+    }
+
+    private String getHeaders(HttpServletRequest request) {
+        Enumeration<String> headerNames = request.getHeaderNames();
+        StringBuilder headers = new StringBuilder();
+        while (headerNames.hasMoreElements()) {
+            String headerName = headerNames.nextElement();
+            String headerValue = request.getHeader(headerName);
+            headers.append(headerName).append(": ").append(headerValue).append("\n");
```

Comment:
Методы в цепочке вызовов лучше располагать на отдельных строках:
```suggestion
            headers.append(headerName)
                .append(": ")
                .append(headerValue)
                .append("\n");
```
Значение для символа переноса на новую строку лучше получать из `System.lineSeparator()`, т.к. он может меняться в зависимости от платформы.